packages/components/src/components/hds/code-editor/index.hbs

```diff
@@ -0,0 +1,53 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: MPL-2.0
+}}
+
+<div class={{this.classNames}} ...attributes>
+  {{#if (and this.isSetupComplete (or this.hasToolbarButton (has-block)))}}
+    <div class="hds-code-editor__header">
+      <div class="hds-code-editor__header-content">
+        {{yield
+          (hash
+            Title=(component "hds/code-editor/title")
+            Description=(component "hds/code-editor/description")
+            Button=(component "hds/code-editor/button")
+          )
+        }}
+      </div>
+
+      {{#if this.hasToolbarButton}}
+        <div class="hds-code-editor__header-toolbar">
+          {{#if @hasCopyButton}}
+            <Hds::Copy::Button
+              class="hds-code-editor__button hds-code-editor__copy-button"
+              @isIconOnly={{true}}
+              @size="small"
+              @text="Copy"
+              @textToCopy={{@value}}
+            />
+          {{/if}}
+          {{#if @hasExpandButton}}
+            <Hds::CodeEditor::Button
+              class="hds-code-editor__expand-button"
+              @isIconOnly={{true}}
+              @size="small"
+              @icon="maximize"
+              @text="Fullscreen"
+              {{on "click" this.toggleFullscreen}}
+            />
+          {{/if}}
+        </div>
+      {{/if}}
+    </div>
+  {{/if}}
+  <div
+    class="hds-code-editor__editor"
+    {{hds-code-editor value=@value language=@language onInput=@onInput onSetup=this.onSetup}}
+  />
+  {{#unless this.isSetupComplete}}
+    <div class="hds-code-editor__loader">
```

Comment:
Agree on the aria/role additions. I'm not sure we can do the loading state in the showcase unless we want to add another external argument to override the loading state. Right now it's just reflective of the loading task state inside the component.

Reply:
There's a couple examples in the showcase like [dropdown](https://github.com/hashicorp/design-system/blob/5102e27eaa446d1e3d6028caa72148ffecc41d5c/showcase/app/templates/components/dropdown.hbs#L456) where we kinda fake a state by using the hds classes instead of the full component. Could be a way around adding a new external arg?

Reply:
After double checking, there's not a way to do this in the showcase that I can think of. We can pass in an optional override that tells the component it's loading, but it will still display the code editor once it's loaded which doesn't provide a clear indication of what the loading state will look like.

Reply:
Thats ok! Its so short it should be fine